pkg/sql/pgwire/hba/hba.go

```diff
@@ -74,43 +113,103 @@ func (h Entry) GetOptions(name string) []string {
 	return val
 }
 
-func (h Entry) String() string {
+// UserMatches returns true iff the provided username matches the
+// first entry in the User list or if the entry matches all.
```

Comment:
I don't think this comment is accurate. This function looks like it can match any entry, not just the first.